src/content/docs/developer/integrations/tx-lifecycle.md

```diff
@@ -11,16 +11,21 @@ Transactions become immutable and are ultimately permanently recorded on the blo
 
-1. **Transaction creation**: The process begins when a wallet or similar software generates a new transaction.
-2. **Transaction broadcasting**: The transaction is sent out to the Dusk network and broadcast within it.
-3. **Transaction reaches Mempool**: The transaction enters the mempool, awaiting validation (transaction included event).
-4. **Transaction validation**: The transaction is removed from the mempool and executed (transaction removed event).
-5. **Transaction execution**: The transaction is executed (transaction executed event).
-   - **Successful Transaction**: No errors available
-   - **Reverted or Failed Transaction**: Errors available (transaction executed event with error field available). This is contract-specific and indicates a revert (panic) or other error that was returned.
-6. **Block acceptance**: The block containing the transaction is accepted into the blockchain (block accepted event).
-7. **Block confirmation**: The block is confirmed when another block is accepted (block state-change event with state changed to `confirmed`).
-8. **Block finalization**: The block reaches finality, there is nothing more to do and the transaction is completed & finalized. (block state-change event with state changed to `finalized`).
+1. **Creation**: The process begins when a wallet or similar software generates a new transaction.
+2. **Broadcasting**: The transaction is sent out to the Dusk network and broadcast within it.
+3. **Validation**: Each node receiving the transaction verifies its validity it before adding it to the Mempool.
+4. **Inclusion in Mempool**: The transaction is added to the Mempool (*transaction included* event).
+   - If the transaction expires before being added to a block, it is removed from the Mempool (*transaction removed* event).
+   - If the transaction is replaced by another transaction with higher gas price, it is removed from the Mempool (*transaction removed* event)
+5. **Inclusion in candidate block**: A block generator includes the transaction from the Mempool into a candidate block.
+   - If the transaction is discarded during the block generation, it is removed from the Mempool (this event is currently not emitted)
+6. **Acceptance**: The block containing the transaction is accepted into the blockchain (*block accepted* event).
+   1. **Execution**: When accepting the block, the transaction is executed (*transaction executed* event).
+      - **Successful Transaction**: No errors available
+      - **Reverted or Failed Transaction**: Errors available (*transaction executed* event with error field available). This is contract-specific and indicates a revert (panic) or other error that was returned.
+   2. **Removal from Mempool**: The transaction is removed from the mempool (*transaction removed* event).
+7. **Confirmation**: The block is confirmed when another block is accepted (*block state-change* event with state changed to `confirmed`).
```

Comment:
I would link a reference to our Rolling Finality, since "block is confirmed when another block is accepted" is not always true

Reply:
Done